src/ext/default.c

```diff
@@ -209,6 +209,81 @@ static double BM25Scorer(const ScoringFunctionArgs *ctx, const RSIndexResult *r,
+  double ret = idf * f * (k1 + 1) / (f + k1 * (1.0f - b + b * (float)doc_len/avg_doc_len));
+  EXPLAIN(scrExp,
+          "%s: (%.2f = IDF %.2f * (F %.2f * (k1 1.2 + 1)) / (F %.2f + k1 1.2 * (1 - b 0.5 + b 0.5 *"
+          " Doc Len %d / Average Doc Len %.2f)))",
+          term, ret, idf, f, f, doc_len, avg_doc_len);
+  return ret;
+}
+
+/* recursively calculate score for each token, summing up sub tokens */
+static double bm25StdRecursive(const ScoringFunctionArgs *ctx, const RSIndexResult *r,
+                            const RSDocumentMetadata *dmd, RSScoreExplain *scrExp) {
+  static const float b = 0.5f;
+  static const float k1 = 1.2f;
+  double f = (double)r->freq;
+  double ret = 0;
+  if (r->type == RSResultType_Term) {
+    // Compute IDF based on total number of docs in the index and the term's total frequency.
+    double idf = CalculateIDF_BM25(ctx->indexStats.numDocs, r->term.term ? r->term.term->numDocs : 0);
+    ret = CalculateBM25Std(b, k1, idf, f, dmd->len, ctx->indexStats.avgDocLen, scrExp,
+                           r->term.term->str);
+  } else if (r->type & (RSResultType_Intersection | RSResultType_Union | RSResultType_HybridMetric)) {
+    int numChildren = r->agg.numChildren;
+    if (!scrExp) {
+      for (int i = 0; i < numChildren; i++) {
+        ret += bm25StdRecursive(ctx, r->agg.children[i], dmd, NULL);
+      }
+    } else {
+      scrExp->numChildren = numChildren;
+      scrExp->children = rm_calloc(numChildren, sizeof(RSScoreExplain));
+      for (int i = 0; i < numChildren; i++) {
+        ret += bm25StdRecursive(ctx, r->agg.children[i], dmd, &scrExp->children[i]);
+      }
+      EXPLAIN(scrExp, "(Weight %.2f * children BM25 %.2f)", r->weight, ret);
+    }
+    ret *= r->weight;
+  } else if (r->type == RSResultType_Virtual && f && r->weight) {
+    // For wildcard, score should be determined only by the weight
+    // and the document's length (so we set idf and f to be 1).
+    double idf = 1.0;
+    double bm25 = CalculateBM25Std(b, k1, idf, 1, dmd->len, ctx->indexStats.avgDocLen, scrExp,
+                                   "*");
+    ret = r->weight * bm25;
+  } else {
+    // Record is either optional term with no match or non text token.
+    // For optional term with no match - we would expect 0 contribution to the score
+    // (the weight should be set to 0).
+    EXPLAIN(scrExp, "Irrelevant token -> score is 0");
+  }
+
+  return ret;
+}
+
+/* BM25 scoring function */
+static double BM25StdScorer(const ScoringFunctionArgs *ctx, const RSIndexResult *r,
+                         const RSDocumentMetadata *dmd, double minScore) {
+  RSScoreExplain *scrExp = (RSScoreExplain *)ctx->scrExp;
+  double bm25res = bm25StdRecursive(ctx, r, dmd, scrExp);
+  double score = dmd->score * bm25res;
+  strExpCreateParent(ctx, &scrExp);
+
+  EXPLAIN(scrExp, "Final BM25 : words BM25 %.2f * document score %.2f", bm25res,
```

Comment:
```suggestion
  EXPLAIN(scrExp, "Final BM25STD : words BM25STD %.2f * document score %.2f", bm25res,
```

Reply:
I'm not sure what I think about this change... The explain string should be clear and friendly. The "STD" is a constraint to differentiate from the original BM25 that we have, but it Is basically is BM25 score...